src/JustSaying/JustSayingBus.cs

```diff
@@ -115,27 +139,43 @@ private void AddSubscribersToInterrogationResponse(INotificationSubscriberInterr
             publishersByType[topicType] = messagePublisher;
         }
 
-        public void Start(CancellationToken cancellationToken = default)
+        public Task Start(CancellationToken cancellationToken)
         {
             if (cancellationToken.IsCancellationRequested)
             {
-                return;
+                return Task.CompletedTask;
+            }
+
+            if (ConsumerBus != null)
+            {
+                _log.LogWarning("Attempting to start an already running Bus");
+                return ConsumerBus.Run(CancellationToken.None);
             }
 
             lock (_syncRoot)
             {
-                foreach (var regionSubscriber in _subscribersByRegionAndQueue)
+                if (ConsumerBus != null)
                 {
-                    foreach (var queueSubscriber in regionSubscriber.Value)
-                    {
-                        if (queueSubscriber.Value.IsListening)
-                        {
-                            continue;
-                        }
-
-                        queueSubscriber.Value.Listen(cancellationToken);
-                    }
+                    _log.LogWarning("Attempting to start an already running Bus");
+                    return ConsumerBus.Run(CancellationToken.None);
```

Comment:
Duplication of code block.

Reply:
I think I've resolved this in a satisfactory way, take a look and see if you're happy with it.